mathematical-companion/main.tex

```diff
@@ -246,8 +248,9 @@ \subsection{Lagrange Interpolation and Shamir's Secret Sharing\label{sec:sss}}
 for fixed $\beta,\alpha_i\in \mathbb{F}$. Then this fixed affine relation
 \emph{will continue to hold for all derived shares}!
 
-This is not immediately obvious but can be shown by direct computation and
-using the fact that Lagrange interpolation is an affine combination of $f_i$'s.
+This is not immediately obvious but can be shown by interpolating the polynomial
+		\[ q(x) = \sum_i \alpha_i p_i(x) - \beta \]
+which, by the assumption that the affine relation holds among all $f_i = p_i(x)$, must be zero everywhere.
```

Comment:
This is not related directly to this PR, but your statement about preservation of affine relationships

    \[ \sum_i \alpha_i f_i = \beta \]

doesn't seem quite right to capture the fact that our BCH checksums are preserved.  As written the relation suggests that you can make a two character change and maintain the stated affine relationship.

i.e. you have made a true statement, but it is the wrong one.

In my view the correct statement is more about BCH codes, which is that any affine combination of codewords is again a codeword, and lagrange interpolation forms affine combinations.

(As an aside, more generally linear combinations of BCH codewords are again codewords so long as a 0 residue is used, but we don't use a 0 residue and affine combinations is all we need).